src/main/java/ch/njol/skript/expressions/ExprSignText.java

```diff
@@ -32,145 +26,203 @@
 
-/**
- * @author Peter Güttinger
- */
 @Name("Sign Text")
 @Description("A line of text on a sign. Can be changed, but remember that there is a 16 character limit per line (including color codes that use 2 characters each).")
-@Examples({"on rightclick on sign:",
-		"	line 2 of the clicked block is \"[Heal]\":",
-		"		heal the player",
-		"	set line 3 to \"%player%\""})
+@Examples({
+		"on rightclick on sign:",
+		"\tline 2 of the clicked block is \"[Heal]\":",
+		"\t\theal the player",
+		"\tset line 3 of block at {_sign} to \"%player%\""
+})
 @Since("1.3")
 public class ExprSignText extends SimpleExpression<String> {
+
 	static {
 		Skript.registerExpression(ExprSignText.class, String.class, ExpressionType.PROPERTY,
-				"[the] line %number% [of %block%]", "[the] (1¦1st|1¦first|2¦2nd|2¦second|3¦3rd|3¦third|4¦4th|4¦fourth) line [of %block%]");
+			"[the] lines [of %block%]",
+			"[the] line %number% [of %block%]",
+			"[the] (1¦1st|1¦first|2¦2nd|2¦second|3¦3rd|3¦third|4¦4th|4¦fourth) line [of %block%]"
+			);
 	}
-	
+
 	private static final ItemType sign = Aliases.javaItemType("sign");
-	
+
+	private boolean isLines;
 	@SuppressWarnings("null")
 	private Expression<Number> line;
 	@SuppressWarnings("null")
-	private Expression<Block> block;
-	
-	@SuppressWarnings({"unchecked", "null"})
-	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		if (matchedPattern == 0)
-			line = (Expression<Number>) exprs[0];
-		else
-			line = new SimpleLiteral<>(parseResult.mark, false);
-		block = (Expression<Block>) exprs[exprs.length - 1];
-		return true;
-	}
-	
+	private Expression<Block> exprBlock;
+	private boolean delayed;
+
 	@Override
-	public boolean isSingle() {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		isLines = (matchedPattern == 0);
+		if (!isLines)
+			line = matchedPattern == 1 ? (Expression<Number>) exprs[0] : new SimpleLiteral<>(parseResult.mark, false);
+		exprBlock = (Expression<Block>) exprs[exprs.length - 1];
+		delayed = isDelayed == Kleenean.TRUE;
 		return true;
 	}
-	
-	@Override
-	public Class<? extends String> getReturnType() {
-		return String.class;
-	}
-	
+
 	@Override
 	@Nullable
-	protected String[] get(final Event e) {
-		final Number l = line.getSingle(e);
-		if (l == null)
-			return new String[0];
-		final int line = l.intValue() - 1;
-		if (line < 0 || line > 3)
-			return new String[0];
-		if (getTime() >= 0 && block.isDefault() && e instanceof SignChangeEvent && !Delay.isDelayed(e)) {
-			return new String[] {((SignChangeEvent) e).getLine(line)};
+	protected String[] get(Event event) {
+		Block block = exprBlock.getSingle(event);
+		if (isLines) {
+			if (block == null || !sign.isOfType(block))
+				return new String[0];
+			if ((event instanceof SignChangeEvent && getTime() >= 0 && block.equals(((SignChangeEvent) event).getBlock()) && !delayed))
```

Comment:
As Skript 2.7 supports [plural event values](https://github.com/SkriptLang/Skript/commit/495fd4f6851a3a968a1818a1aad7441fb0eb5540) which claims from PR author that SignChangeEvent will receive support for this, should the SignChangeEvent support be removed here? Not completely deprecating from being used in SignChangeEvent, but using on the said event will return the lines from the sign's previous state instead.

Reply:
Retaining current support of SignChangeEvent to prevent breaking change.

Reply:
I find it better to include those rather than skipping for breaking change, also could you explain to me how it would become a breaking change? Never looked much into them but I don't fully see how they could cause one.

Reply:
Tud’s plural event value PR will take over this expression for SignChangeEvent, but nevertheless, take this code for example:
```
on sign change:
    broadcast line 1
```
If [SignChangeEvent support](https://github.com/SkriptLang/Skript/blob/d6e0d4c193ed18fc19a8579ed8b8abdff4550281/src/main/java/ch/njol/skript/expressions/ExprSignText.java#L102&L104) is removed, this expression will return nothing as that’s the previous state of the sign.

Reply:
Alright after some mental strain understood this a bit `event-strings`/`line #` return previous state as such I believe it's fine to keep it, for a couple reasons

1. some addons (skbee for example) add a method to open existing signs to players as such they should be updated based off that as the event is called from what I remember.
2. `line 1 of target block` exists so retrieving lines should stay
3. event-strings I don't believe can be changed so `set line 1 to "blah"` should be the only possible method

additionally if a previous type retains then a future state should also tho I'm unsure how that would apply with this part